crates/polars-core/src/frame/column.rs

```diff
@@ -0,0 +1,1187 @@
+use std::borrow::Cow;
+use std::ops::{Add, Div, Mul, Rem, Sub};
+use std::sync::OnceLock;
+
+use num_traits::{Num, NumCast};
+use polars_error::PolarsResult;
+use polars_utils::pl_str::PlSmallStr;
+
+use crate::chunked_array::metadata::MetadataFlags;
+use crate::prelude::*;
+use crate::series::{BitRepr, IsSorted, SeriesPhysIter};
+use crate::utils::Container;
+
+/// A column within a [`DataFrame`].
+///
+/// This is lazily initialized to a [`Series`] with methods like
+/// [`as_materialized_series`][Column::as_materialized_series] and
+/// [`take_materialized_series`][Column::take_materialized_series].
+///
+/// Currently, there are two ways to represent a [`Column`].
+/// 1. A [`Series`] of values
+/// 2. A [`ScalarColumn`] that repeats a single [`Scalar`]
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(from = "Series"))]
+#[cfg_attr(feature = "serde", serde(into = "_SerdeSeries"))]
```

Comment:
This is so that Column serializes and deserializes as if it is a `Series`. Then, we don't have to deal with all the special cases when serializing and deserializing and the DataFrame still looks relatively flat.